lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/VectorShortDecimalFixedWidthByteArrayBatchDecoder.java

```diff
@@ -0,0 +1,371 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader.decoders;
+
+import io.trino.parquet.reader.SimpleSliceInputStream;
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.VectorShuffle;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.parquet.ParquetReaderUtils.propagateSignBit;
+import static io.trino.parquet.reader.decoders.ShortDecimalFixedWidthByteArrayBatchDecoder.BigEndianReader1;
+
+public class VectorShortDecimalFixedWidthByteArrayBatchDecoder
+        implements ShortDecimalDecoder
+{
+    private static final ShortDecimalDecoder[] VALUE_DECODERS = new ShortDecimalDecoder[] {
+            new BigEndianReader1(),
+            new BigEndianReader2(),
+            new BigEndianReader3(),
+            new BigEndianReader4(),
+            new BigEndianReader5(),
+            new BigEndianReader6(),
+            new BigEndianReader7(),
+            new BigEndianReader8()
+    };
+
+    private final ShortDecimalDecoder decoder;
+
+    public VectorShortDecimalFixedWidthByteArrayBatchDecoder(int length)
+    {
+        checkArgument(
+                length > 0 && length <= 8,
+                "Short decimal length %s must be in range 1-8",
+                length);
+        decoder = VALUE_DECODERS[length - 1];
+        // Unscaled number is encoded as two's complement using big-endian byte order
+        // (the most significant byte is the zeroth element)
+    }
+
+    /**
+     * This method uses Unsafe operations on Slice.
+     * Always check if needed data is available with ensureBytesAvailable method.
+     * Failing to do so may result in instant JVM crash.
+     */
+    @Override
+    public void decode(SimpleSliceInputStream input, long[] values, int offset, int length)
+    {
+        decoder.decode(input, values, offset, length);
+    }
+
+    private static final class BigEndianReader8
```

Comment:
Have you tried using VarHandles directly for this case? This is something that the VM supports natively without having to resort to the Vector APIs and may vectorize automatically:

```java
MethodHandles.byteArrayViewVarHandle(long[].class, ByteOrder.BIG_ENDIAN)
```

Under the covers, Slice is already using VarHandles, but the extra indirection, the call to reverseBytes, etc, may be getting in the way of optimizations.

Reply:
Tried VarHandles now, it gives comparable or better performance to vector API, so i've dropped the 2nd commit and added another commit to use VarHandle in the existing implementation instead